package/samplers/dehb/sampler.py

```diff
@@ -0,0 +1,189 @@
+
+        iteration_id = self._get_iteration_id(study, trial)
+        bracket_id = self._get_bracket_id(study, trial)
+        budget_id = self._get_budget_id(study, trial, bracket_id)
+        assert iteration_id > 0
+
+        subpopulation: list[FrozenTrial] = []
+
+        if bracket_id == 0:
+            subpopulation.extend(
+                [
+                    t
+                    for t in trials
+                    if self._get_iteration_id(study, t) == iteration_id - 1
+                    and self._get_bracket_id(study, t) == budget_id
+                    and self._get_budget_id(study, t, bracket_id) == budget_id
+                ]
+            )
+        else:
+            subpopulation.extend(
+                [
+                    t
+                    for t in trials
+                    if self._get_iteration_id(study, t) == iteration_id
+                    and self._get_bracket_id(study, t) == bracket_id - 1
+                    and self._get_budget_id(study, t, bracket_id) == budget_id
+                ]
+            )
+
+        if budget_id > bracket_id:
+            previous_budget_trials = [
+                t
+                for t in trials
+                if self._get_iteration_id(study, t) == iteration_id
+                and self._get_bracket_id(study, t) == bracket_id
+                and self._get_budget_id(study, t, bracket_id) == budget_id - 1
+            ]
+            reduction_factor = study.pruner._pruners[0]._reduction_factor
+            promotable_trials = sorted(
+                previous_budget_trials, key=lambda t: t.intermediate_values[t.last_step]
+            )[: len(previous_budget_trials) // reduction_factor]
+            subpopulation.extend(promotable_trials)
+
+        if len(subpopulation) < 3:
+            subpopulation.extend(trials)
+
+        # Select parents according to the roulette wheel selection.
+        parent0, parent1, parent2 = self._select_parents(subpopulation, study.direction)
+
+        # Mutate selected parents using rand/1 mutation strategy.
+        mutant = self._mutate(parent0, parent1, parent2, search_space)
+
+        # Crossover mutant based on binomial crossover.
+        offspring = self._crossover(mutant, parent0, search_space)
+
+        return offspring
+
+    def sample_independent(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        ret = self._random_sampler.sample_independent(study, trial, param_name, param_distribution)
+        return ret
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        search_space: dict[str, BaseDistribution] = {}
+        for name, distribution in self._search_space.calculate(study).items():
+            if distribution.single():
+                # `cma` cannot handle distributions that contain just a single value, so we skip
```

Comment:
Perhaps should this line be as follows?
```suggestion
                # `DEHBSampler` cannot handle distributions that contain just a single value, so we skip
```

Reply:
I just removed the comments. This comments come from the implementation of Optuna's CmaEsSampler.